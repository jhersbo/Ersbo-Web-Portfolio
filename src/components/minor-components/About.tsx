import './sass/About.scss'
import { motion } from "framer-motion";
import { useFollowPointer } from "../Custom_Hooks/useFollowPointer";
import { useEffect, useRef, useState } from "react";

import AboutContent from "./micro-components/AboutContent";

import CloseIcon from '@mui/icons-material/Close';

const About = ()=>{
    const [clicked, setClicked] = useState(false)
    const [expand, setExpand] = useState(false)
    const [hover, setHover] = useState(false)
    
    const ref = useRef(null);
    const coords = useFollowPointer(ref);
    const previousRef: React.MutableRefObject<{x:number, y:number} | undefined> = useRef()

    const handleClick = ()=>{
        previousRef.current = coords
        setClicked(!clicked)
    }

    useEffect(()=>{
        if(!expand){
            setHover(false)
        }
    }, [expand])

    if(!expand){
        return(
            <motion.div
            ref={ref}
            className="box"
            animate={clicked ? coords : previousRef.current}
            style={{width: "150px", height: "150px"}}
            whileHover={{scale: 1.1,}}
            whileTap={{scale: 0.8}}
            transition={{
                type: "spring",
                damping: 10,
                stiffness: 40,
                restDelta: 0.001
            }}>
                <h3 onClick={()=>{handleClick()}}>About Me</h3>
                <button onClick={()=>{setExpand(!expand)}}>Button</button>
            </motion.div>
        )
    }else{
        return(
            <motion.div
            ref={ref}
            className="expanded-box"
            animate={{x: 0, y: 0, height: "30vh", width: "75vw"}}
            transition={{
                type: "spring",
                damping: 10,
                stiffness: 40,
                restDelta: 0.001
            }}>
                <div className='content-container'>
                    <button 
                    onMouseEnter={()=>{setHover(true)}} 
                    onMouseLeave={()=>{setHover(false)}} 
                    onClick={()=>{setExpand(!expand)}} 
                    style={{
                        border: "10px outset grey", 
                        borderRadius: "50%", 
                        background: hover ? "grey" : "none", 
                    }}>
                        <CloseIcon sx={{fontSize: "36px"}}/>
                    </button>
                    <AboutContent/>
                </div>
            </motion.div>
        )
    }
}

export default About