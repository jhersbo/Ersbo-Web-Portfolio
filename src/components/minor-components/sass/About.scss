.box {
    width: 150px;
    height: 150px;
    border-radius: 50%;
    border: 1px solid red;
  
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.expanded-box{
    border: 1px solid red;
    margin-left: 3em;
    padding: 0.25em 3em;
    border-radius: 10% / 80%;
}

.content-container{
    margin: 1em;
}