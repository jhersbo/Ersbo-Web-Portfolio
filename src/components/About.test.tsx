import About from './About'
import renderer from 'react-test-renderer'

test('passes snapshot test', () => {
  const tree = renderer.create(<About/>).toJSON()
  expect(tree).toMatchSnapshot()
});