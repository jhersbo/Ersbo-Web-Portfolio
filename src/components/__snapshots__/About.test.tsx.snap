// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot test 1`] = `
<div>
  <h1>
    About Me
  </h1>
</div>
`;
